arraylist/src/main/java/ua/hillel/ArrayList.java

```diff
@@ -1,68 +1,125 @@
 package ua.hillel;
 
+import java.util.Objects;
+
 /**
  * Implement a list that is based on array. This resizable data structure
  * is based on an array and is a simplified version of {@link java.util.ArrayList}.
  */
 public class ArrayList<T> implements List<T> {
 
+    private Object[] elements;
+
     @Override
     public void add(T element) {
-
+        if (elements.length == size()) {
+            Object[] newElements = new Object[size() * 2];
+            System.arraycopy(element, 0, newElements, 0, size());
+            elements = newElements;
+        }
+        elements[size()] = element;
```

Comment:
Once you added an element, the size of a list should be changed. You don't have size at all. Create this variable and increment it, if an element was added or decrement, if it was removed.